File: plugins/js/pluginLoader.js
// SYNTHOS Agent Plugin Loader (JS/TS)
// Usage: import plugins dynamically and register them with the agent

export class Agent {
  constructor() {
    this.plugins = [];
    this.capabilities = [];
  }

  async loadPlugin(pluginModule) {
    if (typeof pluginModule.register === 'function') {
      await pluginModule.register(this);
      this.plugins.push(pluginModule);
      if (pluginModule.capabilities) {
        this.capabilities.push(...pluginModule.capabilities);
      }
    }
  }

  listCapabilities() {
    return this.capabilities;
  }
}

// Example usage:
// import * as negotiation from './negotiation.js';
// const agent = new Agent();
// await agent.loadPlugin(negotiation);
// console.log(agent.listCapabilities());

File: plugins/go/pluginloader.go
// SYNTHOS Agent Plugin Loader (Go)
// Usage: import plugins and register them with the agent
package plugins

type Agent struct {
	Plugins      []AgentPlugin
	Capabilities []string
}

type AgentPlugin interface {
	Register(agent *Agent) error
	Capabilities() []string
}

func (a *Agent) LoadPlugin(p AgentPlugin) error {
	err := p.Register(a)
	if err != nil {
		return err
	}
	a.Plugins = append(a.Plugins, p)
	a.Capabilities = append(a.Capabilities, p.Capabilities()...)
	return nil
}

func (a *Agent) ListCapabilities() []string {
	return a.Capabilities
}
